### ISSUE TITLE
`manual_let_else` recommends invalid syntax involving pair pattern binding

### ISSUE THREAD

**Issue by spearman** [rust-lang/rust-clippy#17285]
### Description

The lint warning shows the form of the original code and the recommendation:
```
warning: this could be rewritten as `let...else`
   --> src/collision/mod.rs:694:9
    |
694 | /         let (object_pair, contact) = match constraint {
695 | |           contact@(_, constraint::Variant::Contact (_)) => contact,
696 | |           _ => continue
697 | |         };
    | |__________^ help: consider writing: `let (object_pair, contact) @ (_, constraint::Variant::Contact(_)) = constraint else { continue };`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_let_else
    = note: requested on the command line with `-W clippy::manual-let-else`
```
The recommended code results in the error:
```
error: left-hand side of `@` must be a binding
   --> src/collision/mod.rs:694:13
    |
694 |         let (object_pair, contact) @ (_, constraint::Variant::Contact(_)) = constraint else { continue };
    |             ----------------------^^^------------------------------------
    |             |                        |
    |             |                        also a pattern
    |             interpreted as a pattern, not a binding
    |
    = note: bindings are `x`, `mut x`, `ref x`, and `ref mut x`
```
I believe a correct refactor would be:
```
let (object_pair, contact@constraint::Variant::Contact(_)) = constraint else { continue };
```

### Version

```text
rustc 1.98.0-nightly (8b6558a02 2026-06-20)
binary: rustc
commit-hash: 8b6558a02b2774acfb25cf15e199467c37ba7490
commit-date: 2026-06-20
host: x86_64-unknown-linux-gnu
release: 1.98.0-nightly
LLVM version: 22.1.7
```

**Comment by CommanderStorm** [comment]
Reproduction

```rust
#![warn(clippy::manual_let_else)]
#![expect(unreachable_patterns, dead_code)]

enum Variant {
    Contact(i32),
}

fn main() {
    let constraints = vec![(1, Variant::Contact(42))];

    for constraint in constraints {
        let (object_pair, contact) = match constraint {
            contact @ (_, Variant::Contact(_)) => contact,
            _ => continue,
        };
        let _ = (object_pair, contact);
    }
}
``
